### ISSUE TITLE
Running tutorials on MacOS results in error.

### ISSUE THREAD

**Issue by NiklasGustafsson** [dotnet/TorchSharp#1424]
Trying to run tutorial1.ipynb on MacOS-Arm64 

https://github.com/dotnet/TorchSharpExamples/blob/main/tutorials/CSharp/tutorial1.ipynb

results in an error on the first TorchSharp invocation. It's dynamically looking for the wrong libtorch version number, '2.5.1.0' instead of '2.5.1'

```
Installed Packages
libtorch-cpu-osx-arm64, 2.5.1
TorchSharp-cpu, 0.105.0

Error: System.TypeInitializationException: The type initializer for 'TorchSharp.torch' threw an exception.
---> System.NotSupportedException: The libtorch-cpu-osx-arm64 package version 2.5.1.0 is not restored on this system. If using F# Interactive or .NET Interactive you may need to add a reference to this package, e.g.
#r "nuget: libtorch-cpu-osx-arm64, 2.5.1.0". Trace from LoadNativeBackend:

TorchSharp: LoadNativeBackend: Initialising native backend, useCudaBackend = False
```

This only seems to happen on MacOS, running tutorials on Windows works fine.

**Comment by NiklasGustafsson** [comment]
No, you have to go farther down the notebook, when it first creates a tensor. Before then, nothing is run.

I believe it has to do with .NET Interactive, and how TorchSharp loads the backend dynamically.

**Comment by GeorgeS2019** [comment]
More people with Mac need to check

**Comment by KevinRansom** [comment]
Here is the full stack:
<img width="1440" alt="Image" src="https://github.com/user-attachments/assets/b3dcc2da-40ac-468a-8184-897f086585cb" />

**Comment by KevinRansom** [comment]
Here:   
https://github.com/dotnet/TorchSharp/blob/3760ba3e97ca09a35ad8eab1d5128b62d6d2c2ef/src/TorchSharp/Torch.cs#L24

and here:
https://www.nuget.org/packages/libtorch-cpu-osx-arm64

**Comment by alinpahontu2912** [comment]
Hey @KevinRansom  thanks for the very fast repky. I tried correcting the hardcoded libtorch version in the code and it still failed. I opened the issue on dotnet interactive because when I try to use the packages in a non notebook project (like a console application in visual studio) this error does not happen.

**Comment by alinpahontu2912** [comment]
Will be solved by https://github.com/dotnet/TorchSharp/pull/1461
